dashboard/controller/assignments/addAssignment.js

```diff
@@ -0,0 +1,136 @@
+        req.body.options = { sync: true, stats: true };
+        // get errors
+        var errors = req.validationErrors();
+
+        // call
+        if (!errors) {
+            superagent
+                .post(common.getAPIUrl(req) + 'api/v1/assignments')
+                .set(common.getHeaders(req))
+                .send({
+                    assignment: JSON.stringify(req.body)
+                })
+                .end(function (error, response) {
+                    if (response.statusCode == 200) {
+                        // send to classrooms page
+                        req.flash('success', {
+                            msg: common.l10n.get('AssignmentCreated', { name: req.body.name })
+                        });
+                        return res.redirect('/dashboard/assignments/');
+                    } else {
+                        req.flash('errors', {
+                            msg: common.l10n.get('ErrorCode' + response.body.code)
+                        });
+                        return res.redirect('/dashboard/assignments/add');
+                    }
+                });
+        } else {
+            dashboard_utils.getAllClassrooms(req, res, function (classrooms) {
+                journal_utils.getJournalEntries(req, res, query, function (journalEntries) {
+                    journal_utils.getActivities(req, res, function (activities) {
+                        //make iconMap
+                        var iconMap = {};
+                        for (var i = 0; i < activities.length; i++) {
+                            iconMap[activities[i].id] = '/' + activities[i].directory + '/' + activities[i].icon;
+                        }
+                        if (req.body.classrooms && typeof (req.body.classrooms) == "object" && req.body.classrooms.length > 0 && classrooms && classrooms.classrooms && classrooms.classrooms.length > 0) {
+                            for (var i = 0; i < classrooms.classrooms.length; i++) {
+                                if (req.body.classrooms.indexOf(classrooms.classrooms[i]._id) != -1) {
+                                    classrooms.classrooms[i]['is_member'] = true;
+                                }
+                            }
+                        }
+                        req.flash('errors', errors);
+                        return res.render('addEditAssignment', {
+                            mode: "add",
+                            module: 'assignments',
+                            xocolors: xocolors,
+                            emoji: emoji,
+                            moment: moment,
+                            entries: journalEntries.entries,
+                            classrooms: classrooms.classrooms,
+                            iconMap: iconMap,
+                            account: req.session.user,
+                            server: assignment.ini().information
+                        });
+                    });
```

Comment:
The form data should not be load if there is an error and page reloads.
Test other forms to check the expected behavior.